txtorcon/endpoints.py

```diff
@@ -677,7 +677,6 @@ def __init__(self, host, port,
             except KeyError:
                 pass
 
-        if self.socks_endpoint is None:
             self._socks_port_iter = iter(self.socks_ports_to_try)
```

Comment:
This isn't actually a duplicate: socks_endpoint might still be None here, if a) the user didn't pass socks_endpoint to begin with and b) also didn't use the old style socks_host/socks_port options... (i.e. if it hits the KeyError).